### DISCUSSION TITLE
What needs to be done for vue support?

### DISCUSSION THREAD

**Discussion by pyrareae** [ruby2js/ruby2js#125] [General]
I am seeing in the docs that the Vue filter isn't finished or recommended. What's missing and what needs to be done? I've been itching to be able to use Ruby with Vue so I'd be interested in helping out with that.

**Comment by rubys** [comment]
Brief history of the view filter: the ASF has a number of applications that were using React, but for a while React changed their license to not be open source so the decision was made to move these applications to Vue.  A Vue filter was hastily created to mimic React functions.  That was with Vue 2 at the time.  Since then Vue 3 has been released, and a number of interfaces that are used by the filter are no longer present.

This never was an ideal mapping for Vue.  What should be done instead is for someone with an interest in Vue to define what output they would like to see produced, and then define what the ideal Ruby syntax would be to produce that output.  That could become a test case, and the filter can be updated to pass that test.
